xarray/core/dataset.py

```diff
@@ -521,6 +551,12 @@ def __setitem__(self, key, value):
         if utils.is_dict_like(key):
             raise NotImplementedError('cannot yet use a dictionary as a key '
                                       'to set Dataset values')
+
+        if key in self._level_coords:
+            raise ValueError("%r is already a MultiIndex level of "
```

Comment:
I think this is already checked by `update` (which in turn calls `merge`).

But this is perhaps worth keeping anyways for the better error message? If so, please add a comment